site/app/libraries/DateUtils.php

```diff
@@ -142,4 +142,25 @@ public static function parseDateTimeLong(string $date_time, $default_time_zone =
     public static function dateTimeToString(DateTime $date, bool $add_utc_offset = true) {
         return $date->format('Y-m-d H:i:s' . ($add_utc_offset ? 'O' : ''));
     }
+
+    /**
+     * Gets a json which contains the current server time broken up into specific fields
+     * Formatting the data in this manner makes it easier to work with when instantiating javascript Date() objects
+     *
+     * @param $core Application core
+     * @return object
+     */
+    public static function getServerTimeJson($core) {
+        $time = new \DateTime('now', $core->getConfig()->getTimezone());
+
+        $json = (object)[];
```

Comment:
I am not sure why we have to use `$time->format('x')`-like code for multiple times.

If, for PHP code, put `$time->format("Y-m-j G:i:s")` and return it via JSON, you'll get things like `2019-07-18 1:01:02`, and it's good for JavaScript to accept such a string via `new Date('2019-07-18 1:01:02')`.

Reply:
I believe this is a matter of opinion, but I prefer it this way because the data is more modular.  Having to work between PHP date time formats and javascript date time formats is a nightmare.  Plus if we want to reuse this function in the future maybe we aren't interested in a complete timestamp.  Perhaps on the client side we are only interested in the day of the month and don't want to have to instantiate a javascript Date() object just to get the day back out again.

Reply:
Yeah that makes sense. True that we shouldn't be relying too much on the format consistency between PHP and JavaScript. Thanks for your explanation.

Reply:
It's also important to note that depending on your browser `new Date("datetime string")` gets rendered in different ways with regards to timezones compared to `new Date(year, month, day)`.